File: 4.4 Module VGF/vigo-feast-frontend/src/types/restaurant.ts
export interface Restaurant {
  id: string;
  owner_id: string;
  name: string;
  description: string | null;
  latitude: number | null;
  longitude: number | null;
  phone: string | null;
  email: string | null;
  logo_url: string | null;
  cover_image_url: string | null;
  status: string;
  is_active: boolean;
  is_deleted: boolean;
  created_at: string;
  updated_at: string;
  address_line1: string;
  address_line2: string | null;
  city: string;
  state: string;
  postal_code: string;
}

export interface RestaurantListResponse {
  items: Restaurant[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

File: 4.4 Module VGF/vigo-feast-backend/app/schemas/restaurant.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, EmailStr


class RestaurantBase(BaseModel):
    name: str
    description: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    phone: Optional[str] = None
    email: Optional[EmailStr] = None
    logo_url: Optional[str] = None
    cover_image_url: Optional[str] = None
    status: str
    is_active: bool
    address_line1: str
    address_line2: Optional[str] = None
    city: str
    state: str
    postal_code: str


class RestaurantCreate(RestaurantBase):
    owner_id: str


class RestaurantUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    phone: Optional[str] = None
    email: Optional[EmailStr] = None
    logo_url: Optional[str] = None
    cover_image_url: Optional[str] = None
    status: Optional[str] = None
    is_active: Optional[bool] = None
    address_line1: Optional[str] = None
    address_line2: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    postal_code: Optional[str] = None


class RestaurantResponse(RestaurantBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    owner_id: str
    created_at: datetime
    updated_at: datetime
    is_deleted: bool


class RestaurantListResponse(BaseModel):
    items: list[RestaurantResponse]
    total: int
    page: int
    page_size: int
    total_pages: int
